File: ApiAgendamento/Domain/Entities/Paciente.cs
namespace ApiAgendamento.Domain.Entities;

public class Paciente
{
    public int Id { get; private set; }
    public string Nome { get; private set; }
    public string Email { get; private set; }

    protected Paciente() { }

    public Paciente(string nome, string email)
    {
        Nome = nome;
        Email = email;
    }
}

File: frontend-app/src/models/Paciente.ts
// Modelo (entidade) para Paciente
export interface Paciente {
  id: number;
  nome: string;
  email: string;
}

export interface AgendamentoPaciente {
  id: number
  dataHora: string
  medico: {
    id: number
    nome: string
    especialidade: string
  }
}
